lib/IAMClient.js

```diff
@@ -290,17 +292,25 @@ class VaultClient {
      * @returns {undefined}
      */
     updateAccountQuota(accountName, quota, callback) {
-        const conv = Number.isNaN(Number.parseInt(quota, 10));
-        assert(typeof accountName === 'string' && accountName !== '',
-            'accountName is required');
-        assert(typeof quota === 'number' && !conv
-            && quota >= 0, 'Quota must be a non-negative number');
-        this.request('POST', '/', true, callback, {
+        const disableValidation = this.disableValidation;
+        const data = {
             Action: 'UpdateAccountQuota',
             Version: '2010-05-08',
-            AccountName: accountName,
             quotaMax: quota,
-        });
+        };
+        if (disableValidation) {
+            if (accountName) {
+                data.AccountName = accountName;
+            }
+        } else {
+            const quotaIsValid = typeof quota === 'number' && !Number.isNaN(Number.parseInt(quota, 10)) && quota > 0;
+            assert(quotaIsValid, 'Quota must be a strictly positive number');
+            if (accountName) {
+                assert(typeof accountName === 'string', 'the account name, if set, should be a string');
+                data.AccountName = accountName;
+            }
+        }
```

Comment:
```suggestion
          const quotaIsValid = typeof quota === 'number' && !Number.isNaN(Number.parseInt(quota, 10)) && quota > 0;
          assert(disableValidation || quotaIsValid, 'Quota must be a strictly positive number');
          if (accountName) {
              assert(disableValidation || typeof accountName === 'string', 'the account name, if set, should be a string');
              data.AccountName = accountName;
          }
        }
```

Reply:
actually with this change => whenever the quota is not valid the error will be thrown regardless of the parameter (even if we don't want to check that it is valid or not which is the goal behind the flag)